website/public_adventures.py

```diff
@@ -119,35 +62,85 @@ def filtering(self, user, index_page=False):
                 tags = [_tag for _tag in toReset.split(",") if _tag != tag]
             else:
                 tags = tag.split(",")
+            tags = [t for t in tags if t]
+
+        # Get indexes
+        level_lang_adventure_ids, result = self.db.get_paginated_indexes(
+            {"field_value": f"level_lang_{level}_{language}"},
+            pagination_token=page, limit=20)
+        # In case we need want any language, we retrieve by the main level index.
+        if not level_lang_adventure_ids:
+            level_lang_adventure_ids, result = self.db.get_paginated_indexes({"field_value": f"level_{level}"},
+                                                                             pagination_token=page, limit=20)
```

Comment:
This would all be great detail to hide from the developer in `database.py`!

Together with all the stuff below where we filter on tags and then `batch_get` the remaining adventures. Hiding all that machinery so that the writer of the website can just say `self.db.get_public_adventures()` is the point of that class!